### PR TITLE
Fix the test to not use sigkill

### PR DESCRIPTION
**Pull request by yihuaf**
Sigkill is not the right signal to use because it kills the process without going through the handler. Use sigint instead.

Also in this PR, I decided to refactor the `test_in_child_process` function into a common crate (along with the channel dependency). The `test_in_child_process` function is useful for us because of the process isolation we can use in the test. Cargo runs all tests in the same process, so many of our tests would require process isolation. For example, here we also refactored the `tracing` init related tests using the `test_in_child_process` because we can only call init tracing once.

I called the new crate `common`. I am open to other names. I tried my best separating the crate into `test_utils` and the default feature with the channel implementation, so we can have different dependencies in tests.

### CODE REVIEW

File: crates/common/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+[package]
+name = "common"
```
Comment (utam0k) [inline crates/common/Cargo.toml:2]: All crates must be registered on crates.io in order to register on crates.io.
At least this name cannot be used. I am also not in favor of having a utility-like crate in terms of having to register with crates.io. Why not just use the libcontainer one as public?

Comment (yihuaf) [reply crates/common/Cargo.toml:2]: > All crates must be registered on crates.io in order to register on crates.io.

Good point.

Originally, the `test_utils` is meant to only be used as part of the test with `#[cfg(test)]`. Therefore, I can't share the functions outside of libcontainer with `youki` crate. Perhaps I can create a new `test_utils` feature for the `libcontainer`. Let me try that.

Comment (yihuaf) [reply crates/common/Cargo.toml]: I implemented the `test_utils` as a feature for the libcontainer and set it to `youki` crate's dev dependency. Let me know what you think.

File: crates/libcontainer/src/test_utils.rs [hunk 2]
```diff
@@ -0,0 +1,140 @@
+use nix::sys::wait;
```
[the remark this answers is not in the document]
Comment (yihuaf) [reply crates/libcontainer/src/test_utils.rs:1]: Yes, I set the feature in the `lib.rs`

```
#[cfg(feature = "test_utils")]
pub mod test_utils;
```

I don't think I need to further specify things in the file itself, but I may be wrong.

[the remark this answers is not in the document]
Comment (utam0k) [reply crates/libcontainer/src/test_utils.rs:1]: Oh, sorry, I missed it 😭

**Review by utam0k — APPROVED** [review]

**Pull request merged**